### ISSUE TITLE
"Unsupported browser" error on iPhone with Safari

### ISSUE THREAD

**Issue by ctrlaltdylan** [onfido/onfido-sdk-ui#994]
### What was the expected behaviour?

To be able to upload documents & selfies from an iPhone using Safari with the Onfido UI SDK

### What happened instead?

![Image from iOS](https://user-images.githubusercontent.com/2694734/81592651-4759e680-938c-11ea-83ce-19361ae07cbb.png)

From the JS console on the iPhone:

```
[Warning] getMobileOSName - use of navigator.userAgent is highly discouraged unless absolutely necessary and only for simple use cases (_app.js, line 2)
[Warning] Camera required: Either device has no camera or browser is unable to detect camera (_app.js, line 2)
```

### Version info:

- Onfido SDK v5.7.3
- OS - iOS
- Browser - Safari
- Device - iPhone
- Node - N/A
- Framework or relevant dependencies (if applicable)

### Integration configuration:

```
export default function wizardOptions(user) {
  return {
    language: {
      language: 'en',
      locale: 'en',
      phrases: {
        welcome: {
          next_button: 'Next',
        },
        capture: {
          driving_licence: {
            front: {
              instructions:
                'Upload front of license from your computer. Please use a .jpg, .jpeg, or .png file.',
            },
            back: {
              instructions:
                'Upload back of license from your computer. Please use a .jpg, .jpeg, or .png file.',
            },
          },
          passport: {
            front: {
              instructions:
                'Upload passport photo page from your computer. Please use a .jpg, .jpeg or .png file.',
            },
          },
          face: {
            title:
              'Please take a picture showing your face. For the best results, remove anything that might cover your face such as sunglasses or hats.',
            instructions:
              'Please take a picture showing your face. For the best results, remove anything that might cover your face such as sunglasses or hats.',
          },
        },
        confirm: {
          face: {
            standard: {
              title: 'Check your photo',
              message:
                'Make sure your photo clearly shows your face without anything covering it such as sunglasses or a hat.',
            },
          },
        },
      },
      mobilePhrases: {
        'capture.face.instructions':
          'Please take a picture showing your face. For the best results, remove anything that might cover your face such as sunglasses or hats.',
        'capture.driving_licence.front.instructions':
          "Please take a picture of the front of your driver's license. Be sure the picture is clear and glare free for the best results.",
        'capture.driving_licence.back.instructions':
          "Please take a picture of the back of your driver's license. Be sure the picture is clear and glare free for the best results.",
        'capture.passport.instructions':
          'Please take a picture of your passport. Be sure the picture is clear and glare free for the best results.',
      },
    },
    smsNumberCountryCode: 'US',
    steps: [
      {
        type: 'welcome',
        options: {
          title: `Manually Submit ID Verification on behalf of ${user.full_name}`,
          descriptions: [
            'You can submit the customer\'s photos of their ID document and selfie here for them.',
            'This tool should be used when the customer is unable to submit these documents on their own',
            'It will accept Drivers Licenses, Passports and National ID Cards'
          ],
        },
      },
      {
        type: 'document',
        options: {
          uploadFallback: true,
          initialCrossDeviceDeliveryOption: 'sms',
          documentTypes: {
            passport: true,
            driving_licence: true,
            national_identity_card: true,
          },
        },
      },
      {
        type: 'face',
        options: {
          useWebcam: false,
          uploadFallback: true
        },
      },
    ],
  };
}
```

### Steps to reproduce:

Begin the flow by sending an SMS link to your iPhone. Go through the verification steps, the error occurs on the **selfie** capture step, not the document step - at least for us.

**Comment by stefaniacardenas** [comment]
Hi @ctrlaltdylan, thanks for reporting this.
There is currently [a bug](https://bugs.webkit.org/show_bug.cgi?id=209580) in Safari on  13.1, that has already been reported and fixed, and will be hopefully shipped in the next Safari releases. 
The issue was initially only affecting Safari on Desktop but in the last few days it's affecting iOS too.
We are looking into a possible solution to mitigate this problem while Safari releases a fix.
We have prioritised this issue and will post updates here as soon as we have more information to share.

**Comment by ctrlaltdylan** [comment]
@stefaniacardenas we have set `face.options.uploadFallback` == `true` and it seemed to help in the 3 QA tests we did with iPhone & Safari. Please let us know if we need to update the SDK for the real fix, we're still using the forked version with the configurable option:

https://github.com/onfido/onfido-sdk-ui/pull/900

**Comment by kopijunkie** [comment]
Hi @ctrlaltdylan 

We've just released [5.9.1](https://github.com/onfido/onfido-sdk-ui/releases/tag/5.9.1) with the fix to mitigate this problem in Safari 13.1. Do let us know if you're still seeing issues on iPhone Safari after updating your fork.

**Comment by ctrlaltdylan** [comment]
Thanks will merge and verify

**Comment by ctrlaltdylan** [comment]
@kopijunkie just to clarify, does this mean we'll have to still set `uploadFallback: false` in the `face` option? Or can we revert it back to `false`? It was more secure when bad actors weren't able to upload their own photo.

**Comment by kopijunkie** [comment]
You should be fine to revert back to `false` with this fix now.

**Comment by andra-radu** [comment]
Hi @ctrlaltdylan,

We've just released JS SDK 5.9.1 that has addresses this issue on Safari in iOS 13.4.1
https://github.com/onfido/onfido-sdk-ui/releases/tag/5.9.1

**Comment by kopijunkie** [comment]
As a follow up, unfortunately there was a bug in that release's workaround that causes a slight delay for the Document step loading on non-Safari browsers. Please update your fork to [5.9.2](https://github.com/onfido/onfido-sdk-ui/releases/tag/5.9.2) instead.
